cli/server.go

```diff
@@ -723,6 +723,17 @@ func (r *RootCmd) Server(newAPI func(context.Context, *coderd.Options) (*coderd.
 						return xerrors.Errorf("register agent stats prometheus metric: %w", err)
 					}
 					defer closeAgentStatsFunc()
+
+					metricsAggregator, err := prometheusmetrics.NewMetricsAggregator(logger, options.PrometheusRegistry, 0)
+					if err != nil {
+						return xerrors.Errorf("can't initialize metrics aggregator: %w", err)
+					}
+
+					cancelMetricsAggregator := metricsAggregator.Run(ctx)
+					defer cancelMetricsAggregator()
+
+					options.UpdateAgentMetrics = metricsAggregator.Update
+					options.PrometheusRegistry.MustRegister(metricsAggregator)
```

Comment:
Is there a non-must method? We could probably give a nicer error than e.g. a panic (assuming that's what it does) if we check it manually.

Reply:
Good call, fixed!